src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Quic.Tests
+{
+    public class MsQuicTests : MsQuicTestBase
+    {
+        private static ReadOnlyMemory<byte> s_data = Encoding.UTF8.GetBytes("Hello world!");
+
+        [Fact]
+        public async Task BasicTest()
+        {
+            Task listenTask = Task.Run(async () =>
+            {
+                // Right now, AcceptConnections is required to start before staring a client connection.
+                // We can try to fix this by either 
+                using (QuicConnection connection = await DefaultListener.AcceptConnectionAsync())
+                {
+                    using (QuicStream stream = await connection.AcceptStreamAsync())
+                    {
+                        byte[] buffer = new byte[s_data.Length];
+                        int bytesRead = await stream.ReadAsync(buffer);
+                        Assert.Equal(s_data.Length, bytesRead);
+                        Assert.True(s_data.Span.SequenceEqual(buffer));
+                        await stream.WriteAsync(s_data);
+                    }
+                }
+            });
+
+            Task clientTask = Task.Run(async () =>
+            {
+                await using (QuicConnection connection = CreateQuicConnection(DefaultEndpoint))
+                {
+                    await connection.ConnectAsync();
+                    using (QuicStream stream = connection.OpenBidirectionalStream())
+                    {
+                        await stream.WriteAsync(s_data);
+                        byte[] memory = new byte[12];
+                        int res = await stream.ReadAsync(memory);
+                        Assert.True(s_data.Span.SequenceEqual(memory));
+                    }
+                }
+            });
+
+            await Task.WhenAll(listenTask, clientTask);
```

Comment:
Use `new[]{listenTask, clientTask}.WhenAllOrAnyFailed(timeoutMs)` rather than `WhenAll` to avoid stalling tests on error.